modules/3d/include/opencv2/3d/volume_settings.hpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+
+#ifndef OPENCV_3D_VOLUME_SETTINGS_HPP
+#define OPENCV_3D_VOLUME_SETTINGS_HPP
+
+#include <opencv2/core/cvstd.hpp>
+#include <opencv2/3d/volume.hpp>
+
+namespace cv
+{
+
+enum class VolumeType
+{
+    TSDF = 0,
+    HashTSDF = 1,
+    ColorTSDF = 2
+};
+
+
+class CV_EXPORTS_W VolumeSettings
+{
+public:
+    VolumeSettings();
+    VolumeSettings(VolumeType volumeType);
+    ~VolumeSettings();
+
+    void  setWidth(int val);
+    int   getWidth() const;
+    void  setHeight(int val);
+    int   getHeight() const;
+    void  setDepthFactor(float val);
+    float getDepthFactor() const;
+    void  setVoxelSize(float val);
+    float getVoxelSize() const;
+    void  setTruncatedDistance(float val);
+    float getTruncatedDistance() const;
+    void  setTruncateThreshold(float val);
+    float getTruncateThreshold() const;
+    void  setMaxWeight(int val);
+    int   getMaxWeight() const;
+    void  setRaycastStepFactor(float val);
+    float getRaycastStepFactor() const;
+    void  setZFirstMemOrder(bool val);
+    bool  getZFirstMemOrder() const;
+
+    void setVolumePose(InputArray val);
+    void getVolumePose(OutputArray val) const;
+    void setVolumeResolution(InputArray val);
+    void getVolumeResolution(OutputArray val) const;
+    void getVolumeDimentions(OutputArray val) const;
+    void setCameraIntegrateIntrinsics(InputArray val);
```

Comment:
For this and the following functions:
Please add a doc about `set/get Camera Intrinsics` that they are about the depth camera.